Tests/LibC/TestMkDir.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021, Max Wipfli <max.wipfli@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+
+#include <errno.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <time.h>
+#include <unistd.h>
+
+static String random_dirname()
+{
+    return String::formatted("/tmp/test_mkdir_{:04x}", (u16)rand());
+}
+
+TEST_SETUP
+{
+    srand(time(NULL));
+}
+
+TEST_CASE(basic)
+{
+    auto dirname = random_dirname();
+    int res = mkdir(dirname.characters(), 0755);
+    EXPECT(res == 0);
+
+    res = mkdir(dirname.characters(), 0755);
+    int cached_errno = errno;
+    EXPECT(res < 0);
+    EXPECT_EQ(cached_errno, EEXIST);
+}
+
+TEST_CASE(insufficient_permissions)
+{
+    VERIFY(getuid() != 0);
+    int res = mkdir("/root/foo", 0755);
+    int cached_errno = errno;
+    EXPECT(res < 0);
+    EXPECT_EQ(cached_errno, EACCES);
+}
+
+TEST_CASE(nonexistent_parent)
+{
+    auto parent = random_dirname();
+    auto child = String::formatted("{}/foo", parent);
+    int res = mkdir(child.characters(), 0755);
+    int cached_errno = errno;
+    EXPECT(res < 0);
+    EXPECT_EQ(cached_errno, ENOENT);
+}
+
+TEST_CASE(parent_is_file)
+{
+    int res = mkdir("/etc/passwd/foo", 0755);
+    int cached_errno = errno;
+    EXPECT(res < 0);
+    EXPECT_EQ(cached_errno, ENOTDIR);
+}
+
+TEST_CASE(pledge)
+{
+    int res = pledge("stdio cpath", nullptr);
+    EXPECT(res == 0);
+
+    auto dirname = random_dirname();
+    res = mkdir(dirname.characters(), 0755);
+    EXPECT(res == 0);
+    // FIXME: Somehow also check that mkdir() stops working when removing the cpath promise. This is currently
```

Comment:
I think you could do this with the `EXPECT_CRASH(..)` or `EXPECT_CRASH_WITH_SIGNAL(..)` test, since they fork and run in a separate process?

Reply:
See the discussion in #8643. Basically, pledge/unveil restrictions are only removed on exec(), not on fork(). Thus, `EXCEPT_CRASH` and friends do not help in this context (as far as I am aware). I have tested this when we talked about it in #8643.

Reply:
Ah yes, I vaguely recall that discussion. Sorry for rehashing the same ole thing :smile: